packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/OutputTypes/ScmKnownParameters.cs

```diff
@@ -30,12 +30,12 @@ internal static class ScmKnownParameters
         public static readonly ParameterProvider Data = new ParameterProvider("data", FormattableStringHelpers.Empty, typeof(BinaryData));
 
         private static ParameterProvider? _tokenAuth;
-        public static ParameterProvider TokenAuth => _tokenAuth ??= new("tokenCredential", $"The token credential to copy", ClientModelPlugin.Instance.TypeFactory.TokenCredentialType());
+        public static ParameterProvider TokenAuth => _tokenAuth ??= new("tokenCredential", $"The token credential to copy", ((ScmTypeFactory)CodeModelPlugin.Instance.TypeFactory).TokenCredentialType());
```

Comment:
this casting is unfortunate, we should probably rethink how we want this to look

Reply:
Will file a follow up issue.

Reply:
https://github.com/microsoft/typespec/issues/3701